spec/components/schools/teacher_profile_summary_list_component_spec.rb

```diff
@@ -0,0 +1,43 @@
+require "rails_helper"
+
+RSpec.describe Schools::TeacherProfileSummaryListComponent, type: :component do
+  include TeacherHelper
+  include ECTHelper
+
+  let(:ect) { FactoryBot.create(:ect_at_school_period, :active, started_on: 3.years.ago) }
+  let(:mentor) { FactoryBot.create(:mentor_at_school_period, :active, started_on: 3.years.ago) }
+  let!(:ongoing_mentorship) { FactoryBot.create(:mentorship_period, :active, mentee: ect, mentor:) }
+
+  before do
+    render_inline(described_class.new(ect))
+  end
+
+  it "renders the summary list container" do
+    expect(page).to have_selector(".govuk-summary-list")
+  end
+
+  it "renders the name row with correct value" do
+    expect(page).to have_selector(".govuk-summary-list__row", text: "Name")
+    expect(page).to have_text(teacher_full_name(ect.teacher))
+  end
+
+  it "renders the email row with correct value" do
+    expect(page).to have_selector(".govuk-summary-list__row", text: "Email address")
+    expect(page).to have_text(ect.email)
+  end
+
+  it "renders the mentor row with correct value" do
+    expect(page).to have_selector(".govuk-summary-list__row", text: "Mentor")
+    expect(page).to have_text(ect_mentor_details(ect))
```

Comment:
I know we aren't always doing this in other specs but I am an advocate of asserting the raw string (or partial string). As in, not including the helpers, factorying a `Teacher` with known values, and using `have_text("this is the text")`.

Yes it does mean that changes to the helper methods output can break here, but I like that

Reply:
I usually like asserting against raw strings in tests too but on this occasion we are using Faker to generate things dynamically, are you suggesting that we hard code the name into the factory instead and then assert against it?

For example: https://github.com/DFE-Digital/register-early-career-teachers-public/blob/main/spec/factories/teacher_factory.rb#L4-L5

Reply:
```ruby
let(:teacher) { FactoryBot.create(:teacher, trn: '9876543', trs_first_name: 'Avin', trs_last_name: 'Hurry') }
let(:ect) { FactoryBot.create(:ect_at_school_period, :active, teacher:, started_on: 3.years.ago) }
```

I thought instead we could define the `Teacher` the ECT is linked to and gets the name from. If a spec also cares about the mentor you need to make two `Teacher` records with this approach, and there are examples in the suite where we do this with `mentor` and `mentee` as the variable names.

Reply:
Ah yes that makes sense, thanks! Happy to do it that way if thats the approach we prefer but it would be nice to be consistent going forward.

I've updated it in this commit: https://github.com/DFE-Digital/register-early-career-teachers-public/pull/313/commits/a14d776d08f66cdef1146b67cd149eae377d2da9